### PR TITLE
make propagationPolicy configurable during job rollout

### LINKED ISSUE #2910: ScaledJob Scaler does not take "Terminating" Pods into consideration
### Report

When a new scaledJob spec is rolled out, all existing Pods for those jobs receive a Sig-Term when using the default rollout strategy. This is desired behavior for us and fits our usecase.

Since we are doing long term calculations in those pods, we ignore the Sig-Term and our pod finishes its work. The Pod status of those sig-termed Pods is "Terminating".

Keda only seems to care about Pods that are either Pending or Running. As a consequence Keda will not count the "Terminating" Pods to the already running and healthy Pods. This means the amount of created Jobs by Keda is off by the number of Pods that are in state "Terminating".

Is there a possibility to configure that "Terminating" Pods are counted as running Pods as well? We had a look into the PendingPodConditions but they dont seem to work with the actual Pod status (Terminating).

### Expected Behavior

KEDA should create N pods - number of the Terminating pods from the previous version.

### Actual Behavior

KEDA will create N pods + number of the Terminating pods from the previous version.

### Steps to Reproduce the Problem

1. Delay Sig-Term Shutdown in your workload pod (the one that gets created by the ScaledJob)
2. Rollout new ScaledJob-Spec or send a sig-Term to one of your workload pods
3. Number of created Jobs is off by the number of "Terminating" Pods

### KEDA Version

2.5.0

### Kubernetes Version

1.21

### Platform

Microsoft Azure

### Scaler Details

Redis

### Anything else?

Slack-Discussion: https://kubernetes.slack.com/archives/CKZJ36A5D/p1649865034815469

### PR DESCRIPTION
**Pull request by FullyScaled**
Signed-off-by: Jonas Bentz <jonas.bentz@sap.com>

This PR allows to configure the kubernetes propagation policy during job scaled roll out. 
In some usecases (see Issue https://github.com/kedacore/keda/issues/2910) its required to use foreground propagation instead of background propagation during rollout. 

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2910

Relates to # https://github.com/kedacore/keda-docs/pull/804

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]

File: controllers/keda/scaledjob_controller.go [hunk 1]
```diff
@@ -187,7 +195,12 @@ func (r *ScaledJobReconciler) deletePreviousVersionScaleJobs(ctx context.Context
 		}
 		for _, job := range jobs.Items {
 			job := job
-			err = r.Client.Delete(ctx, &job, client.PropagationPolicy(metav1.DeletePropagationBackground))
+
+			propagationPolicy := metav1.DeletePropagationBackground
+			if scaledJob.Spec.Rollout.PropagationPolicy == "foreground" {
```
[the remark this answers is not in the document]
Comment (FullyScaled) [reply controllers/keda/scaledjob_controller.go:200]: Unfortunately not, I can add one though if this is desired.

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**